src/supermarket/spec/controllers/api/v1/cookbook_uploads_controller_spec.rb

```diff
@@ -244,6 +244,21 @@
       end
     end
 
+    context 'when there is only one version of the cookbook remaining' do
+      before { auto_authorize!(Cookbook, 'destroy') }
+
+      it 'returns a 409 with informative error messages' do
+        versions = CookbookVersion.all.pluck(:version)
+        versions.each do |version|
+          delete :destroy_version, cookbook: cookbook, version: version, format: :json
+        end
+
+        expect(response.status.to_i).to eql(409)
+        expect(JSON.parse(response.body)["error_code"]).to_not eq(nil)
+        expect(I18n.t('api.error_messages.only_cookbook_version')).to eq('You may not DELETE the only version of a cookbook. A cookbook must have at least one version.')
```

Comment:
I suggest something like
```
expect(JSON.parse(response.body)['error']).to eq(I18n.t('api.error_messages.only_cookbook_version'))
```
 to test that the error displays a labeled error message, but not care about the specific wording of the translation.